timeseries/src/main/scala/com/criteo/cuttle/timeseries/TimeSeriesScheduler.scala

```diff
@@ -370,44 +374,150 @@ case class TimeSeriesScheduler(logger: Logger) extends Scheduler[TimeSeries] wit
+        Future(List.empty)
+      else {
+        val executionsToCancel =
+          listExecutionsToCancelGivenBackfills(validBackfills, _state(), runningExecutions.keys)
+        executionsToCancel.foreach { e =>
+          logger.info(s"Cancelling running execution ${e.id} for job ${e.job.id}")
+          e.cancel()
+        }
+        Future.sequence(
+          executionsToCancel.map(e => runningExecutions.apply(e))
+        )
+      }
+      (validBackfills, futureCancelledExecutions)
+    }
+
+    if (errors.nonEmpty)
+      IO.pure(Left(errors.mkString("\n")))
+    else {
+      futureCancelledExecutions.andThen { case _ => validBackfills.foreach { newBackfill =>
+        updateBackfillState(newBackfill)
+        Database.createBackfill(newBackfill).transact(xa)
+      } }
+      IO.pure(Right(Unit))
+    }
+  }
+
+  private[timeseries] def listExecutionsToCancelGivenBackfills(
+      backfills: List[Backfill],
+      states: Map[TimeSeriesUtils.TimeSeriesJob, IntervalMap[Instant, JobState]],
+      runningExecutions: Iterable[Execution[TimeSeries]]): List[Execution[TimeSeries]] = {
+
+    val intervalsToBackfill = backfills.map(backfill => Interval(backfill.start, backfill.end))
+    val jobsToBackfill = backfills.flatMap(_.jobs)
+
+    lazy val executionsById = runningExecutions.map(e => e.id -> e).toMap
+
+    // collect all the running executions that are intersecting with the backfill period
+    val executionsToCancel = intervalsToBackfill.flatMap { interval =>
+      jobsToBackfill.flatMap { job =>
+        val interval2State = states(job).intersect(interval).toList
+        interval2State.collect {
+          case (period, Running(executionId)) => executionsById(executionId)
+        }
+      }
+    }
+
+    executionsToCancel
+  }
+
+  private[timeseries] def createBackfills(name: String,
+                                          description: String,
+                                          jobs: Set[TimeSeriesJob],
+                                          states: Map[TimeSeriesUtils.TimeSeriesJob, IntervalMap[Instant, JobState]],
+                                          start: Instant,
+                                          end: Instant,
+                                          priority: Int)(implicit user: Auth.User): List[Backfill] = {
+    val interval = Interval(start, end)
+
+    // Find jobs which can be backfilled on the requested period. Those are the jobs whose state is 'Done'.
+    val candidateBackfillsByPeriod: List[((Instant, Instant), TimeSeriesJob)] = jobs.flatMap { job =>
+      // collect all periods that are intersecting with [start, end]
+      val interval2State = states(job).intersect(interval).toList
+      // in a Done state, and correct periodicity
+      interval2State.collect { case (Interval(Finite(lo), Finite(hi)), Done) =>
+        (lo, hi) -> job
+      }
+    }.toList
+
+    val jobsToBackFillByPeriod = candidateBackfillsByPeriod.groupBy { case (interval, job) => interval }
```

Comment:
This wont be able to group correctly. Say for example job1 is backfillable from 1 to 3 & jobs depending of job1 from 1 to 2, they will be created as separate backfills.